### DISCUSSION TITLE
Foldable Component with embeded elements

### DISCUSSION THREAD

**Discussion by DSMejantel** [sqlpage/SQLPage#597] [Q&A]
Hi !
Nice feature. Is it possible to embed other component into the foldable component ?

**Comment by YoannCalamai** [comment]
Hi,

Can you provide an example of what you are trying to achieve?

**Comment by lovasoa** [comment]
Hello ! There is currently no way to embed other components in the foldable component. 

So if you have things like forms and buttons that you want on include inside a foldable, that is currently not possible. That would be a nice thing to add in the future, though.

If your content is purely textual (including rich text like markdown), you can of course still use `sqlpage.run_sql` and the `dynamic` component to generate the data for the foldable.

**Reply by phb170465** [reply]
Hi lovasoa
I would be very interested with a small  example 'how to include dynamic component in foldable'.
I'm quite new in sql-page ;-) and I'm struggling with using dynamic component.

My final purpose should be to build hierarchical table, that means : 

- having table with 2 column (name, department )
- display data grouped by departments
-  each department as foldable component 
- each name in a department, as table component in foldable component

Even if not possible, still interested with example dynamic comp in foldable comp

**Reply by lovasoa** [reply]
If you have a table like `employees(name tex, department text)`, you can display it in the foldable component like that:

```sql
select 'foldable' as component;
select
    department as title,
    (
        'Employees in this department:' || char(10) || char(10) ||
        group_concat('- ' || name, char(10))
    ) as description_md
from employees
group by department
order by department;
```

<img width="2545" height="1241" alt="image" src="https://github.com/user-attachments/assets/be86e15d-32bc-4483-9227-454b8f071f57" />

**Comment by ToniJ123** [comment]
Hi,

Yeah we would have use purposes for this feature also, if there could be embed forms or any other components inside it!

Anyway, great component this foldable, just no use for only text. 👍
